diff -urN -x .build -x .inst -x .sinst -x configure -x 'Makefile.in*' -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x 'mkinstalldirs*' -x autom4te.cache -x '*compile' freetds-0.63-orig/CYGWIN-PATCHES/README freetds-0.63/CYGWIN-PATCHES/README
--- freetds-0.63-orig/CYGWIN-PATCHES/README	1969-12-31 18:00:00.000000000 -0600
+++ freetds-0.63/CYGWIN-PATCHES/README	2005-11-14 17:51:26.796875000 -0600
@@ -0,0 +1,307 @@
+freetds
+------------------------------------------
+FreeTDS is a set of libraries for Unix and Linux that allows your
+programs to natively talk to Microsoft SQL Server and Sybase databases.
+
+Technically speaking, FreeTDS is an open source implementation of the
+TDS (Tabular Data Stream) protocol used by these databases for their own
+clients. It supports many different flavors of the protocol and three
+APIs to access it. Additionally FreeTDS works with other software such
+as Perl and PHP, providing access from those languages as well. 
+
+Runtime requirements:
+  cygwin-1.5.18-1
+  libiconv2-1.9.2-1
+  libncurses8-5.4-4
+  libreadline6-5.0-4
+
+Build requirements:
+  cygwin-1.5.18-1
+  autoconf2.5-2.59-3
+  automake1.9-1.9.6.1
+  binutils-20050610-1
+  bison-20030307-1
+  flex-2.5.4a-3
+  gawk-3.1.5-1
+  gcc-core-3.4.4-1
+  libtool1.5-1.5.20-1
+  make-3.80-1
+  sed-4.1.4-1
+
+Canonical website:
+  http://www.freetds.org/
+
+Canonical download:
+  ftp://ftp.ibiblio.org/pub/Linux/ALPHA/freetds/stable/
+
+-------------------------------------------
+
+Build instructions:
+  unpack freetds-0.63-1-src.tar.bz2
+    if you use setup to install this src package, it will be
+	 unpacked under /usr/src automatically
+  cd /usr/src
+  ./freetds-0.63-1.sh all
+
+This will create:
+  /usr/src/freetds-0.63-1-src.tar.bz2
+  /usr/src/freetds-0.63-1.tar.bz2
+  /usr/src/freetds-devel-0.63-1.tar.bz2
+  /usr/src/freetds-doc-0.63-1.tar.bz2
+  /usr/src/libct3-0.63-1.tar.bz2
+  /usr/src/libsybdb5-0.63-1.tar.bz2
+  /usr/src/libtds4-0.63-1.tar.bz2
+  /usr/src/libtdssrv2-0.63-1.tar.bz2
+
+-------------------------------------------
+
+Files included in the binary package:
+
+  /etc/freetds.conf
+  /etc/locales.conf
+  /etc/pool.conf
+  /usr/bin/bsqldb.exe
+  /usr/bin/defncopy.exe
+  /usr/bin/freebcp.exe
+  /usr/bin/tdspool.exe
+  /usr/bin/tsql.exe
+  /usr/share/doc/Cygwin/freetds-0.63.README
+  /usr/share/doc/freetds-0.63/AUTHORS
+  /usr/share/doc/freetds-0.63/BUGS
+  /usr/share/doc/freetds-0.63/ChangeLog
+  /usr/share/doc/freetds-0.63/COPYING
+  /usr/share/doc/freetds-0.63/COPYING.LIB
+  /usr/share/doc/freetds-0.63/INSTALL
+  /usr/share/doc/freetds-0.63/NEWS
+  /usr/share/doc/freetds-0.63/README
+  /usr/share/doc/freetds-0.63/TODO
+  /usr/share/doc/freetds-0.63/api_status.txt
+  /usr/share/doc/freetds-0.63/bcp.txt
+  /usr/share/doc/freetds-0.63/cap.txt
+  /usr/share/doc/freetds-0.63/getting_started.txt
+  /usr/share/doc/freetds-0.63/policy.txt
+  /usr/share/man/man1/bsqldb.1.gz
+  /usr/share/man/man1/defncopy.1.gz
+  /usr/share/man/man1/freebcp.1.gz
+  /usr/share/man/man1/tsql.1.gz
+
+Files included in the respective lib packages:
+
+  /usr/bin/cygct-3.dll
+
+  /usr/bin/cygsybdb-5.dll
+
+  /usr/bin/cygtds-4.dll
+
+  /usr/bin/cygtdssrv-2.dll
+
+Files included in the devel package:
+
+  /usr/include/freetds/bkpublic.h
+  /usr/include/freetds/cspublic.h
+  /usr/include/freetds/cstypes.h
+  /usr/include/freetds/ctpublic.h
+  /usr/include/freetds/sqldb.h
+  /usr/include/freetds/sqlfront.h
+  /usr/include/freetds/sybdb.h
+  /usr/include/freetds/syberror.h
+  /usr/include/freetds/sybfront.h
+  /usr/include/freetds/tds.h
+  /usr/include/freetds/tdsconvert.h
+  /usr/include/freetds/tdssrv.h
+  /usr/include/freetds/tdsver.h
+  /usr/include/freetds/tds_sysdep_public.h
+  /usr/lib/libct.a
+  /usr/lib/libct.dll.a
+  /usr/lib/libct.la
+  /usr/lib/libsybdb.a
+  /usr/lib/libsybdb.dll.a
+  /usr/lib/libsybdb.la
+  /usr/lib/libtds.a
+  /usr/lib/libtds.dll.a
+  /usr/lib/libtds.la
+  /usr/lib/libtdssrv.a
+  /usr/lib/libtdssrv.dll.a
+  /usr/lib/libtdssrv.la
+
+Files included in the doc package:
+
+  /usr/share/doc/freetds-0.63/html/tds.html
+  /usr/share/doc/freetds-0.63/html/images/important.gif
+  /usr/share/doc/freetds-0.63/html/images/note.gif
+  /usr/share/doc/freetds-0.63/html/images/tip.gif
+  /usr/share/doc/freetds-0.63/html/reference/a00064.html
+  /usr/share/doc/freetds-0.63/html/reference/a00075.html
+  /usr/share/doc/freetds-0.63/html/reference/a00081.html
+  /usr/share/doc/freetds-0.63/html/reference/a00082.html
+  /usr/share/doc/freetds-0.63/html/reference/a00083.html
+  /usr/share/doc/freetds-0.63/html/reference/a00085.html
+  /usr/share/doc/freetds-0.63/html/reference/a00087.html
+  /usr/share/doc/freetds-0.63/html/reference/a00156.html
+  /usr/share/doc/freetds-0.63/html/reference/a00159.html
+  /usr/share/doc/freetds-0.63/html/reference/a00194.html
+  /usr/share/doc/freetds-0.63/html/reference/a00197.html
+  /usr/share/doc/freetds-0.63/html/reference/a00198.html
+  /usr/share/doc/freetds-0.63/html/reference/a00213.html
+  /usr/share/doc/freetds-0.63/html/reference/a00217.html
+  /usr/share/doc/freetds-0.63/html/reference/a00232.html
+  /usr/share/doc/freetds-0.63/html/reference/a00255.html
+  /usr/share/doc/freetds-0.63/html/reference/a00307.html
+  /usr/share/doc/freetds-0.63/html/reference/a00310.html
+  /usr/share/doc/freetds-0.63/html/reference/a00336.html
+  /usr/share/doc/freetds-0.63/html/reference/a00337.html
+  /usr/share/doc/freetds-0.63/html/reference/a00338.html
+  /usr/share/doc/freetds-0.63/html/reference/a00339.html
+  /usr/share/doc/freetds-0.63/html/reference/a00340.html
+  /usr/share/doc/freetds-0.63/html/reference/a00341.html
+  /usr/share/doc/freetds-0.63/html/reference/a00342.html
+  /usr/share/doc/freetds-0.63/html/reference/a00343.html
+  /usr/share/doc/freetds-0.63/html/reference/a00344.html
+  /usr/share/doc/freetds-0.63/html/reference/a00345.html
+  /usr/share/doc/freetds-0.63/html/reference/a00346.html
+  /usr/share/doc/freetds-0.63/html/reference/a00347.html
+  /usr/share/doc/freetds-0.63/html/reference/a00348.html
+  /usr/share/doc/freetds-0.63/html/reference/a00349.html
+  /usr/share/doc/freetds-0.63/html/reference/a00350.html
+  /usr/share/doc/freetds-0.63/html/reference/a00351.html
+  /usr/share/doc/freetds-0.63/html/reference/a00352.html
+  /usr/share/doc/freetds-0.63/html/reference/a00353.html
+  /usr/share/doc/freetds-0.63/html/reference/a00354.html
+  /usr/share/doc/freetds-0.63/html/reference/a00356.html
+  /usr/share/doc/freetds-0.63/html/reference/a00358.html
+  /usr/share/doc/freetds-0.63/html/reference/a00359.html
+  /usr/share/doc/freetds-0.63/html/reference/a00360.html
+  /usr/share/doc/freetds-0.63/html/reference/a00361.html
+  /usr/share/doc/freetds-0.63/html/reference/a00362.html
+  /usr/share/doc/freetds-0.63/html/reference/a00363.html
+  /usr/share/doc/freetds-0.63/html/reference/a00364.html
+  /usr/share/doc/freetds-0.63/html/reference/a00365.html
+  /usr/share/doc/freetds-0.63/html/reference/a00366.html
+  /usr/share/doc/freetds-0.63/html/reference/a00367.html
+  /usr/share/doc/freetds-0.63/html/reference/a00368.html
+  /usr/share/doc/freetds-0.63/html/reference/a00369.html
+  /usr/share/doc/freetds-0.63/html/reference/a00370.html
+  /usr/share/doc/freetds-0.63/html/reference/a00371.html
+  /usr/share/doc/freetds-0.63/html/reference/a00372.html
+  /usr/share/doc/freetds-0.63/html/reference/a00373.html
+  /usr/share/doc/freetds-0.63/html/reference/a00374.html
+  /usr/share/doc/freetds-0.63/html/reference/a00375.html
+  /usr/share/doc/freetds-0.63/html/reference/a00376.html
+  /usr/share/doc/freetds-0.63/html/reference/a00377.html
+  /usr/share/doc/freetds-0.63/html/reference/a00378.html
+  /usr/share/doc/freetds-0.63/html/reference/a00379.html
+  /usr/share/doc/freetds-0.63/html/reference/a00380.html
+  /usr/share/doc/freetds-0.63/html/reference/a00381.html
+  /usr/share/doc/freetds-0.63/html/reference/a00382.html
+  /usr/share/doc/freetds-0.63/html/reference/a00383.html
+  /usr/share/doc/freetds-0.63/html/reference/a00384.html
+  /usr/share/doc/freetds-0.63/html/reference/a00385.html
+  /usr/share/doc/freetds-0.63/html/reference/a00386.html
+  /usr/share/doc/freetds-0.63/html/reference/a00387.html
+  /usr/share/doc/freetds-0.63/html/reference/a00388.html
+  /usr/share/doc/freetds-0.63/html/reference/a00389.html
+  /usr/share/doc/freetds-0.63/html/reference/a00391.html
+  /usr/share/doc/freetds-0.63/html/reference/a00393.html
+  /usr/share/doc/freetds-0.63/html/reference/a00394.html
+  /usr/share/doc/freetds-0.63/html/reference/a00395.html
+  /usr/share/doc/freetds-0.63/html/reference/a00396.html
+  /usr/share/doc/freetds-0.63/html/reference/a00397.html
+  /usr/share/doc/freetds-0.63/html/reference/a00398.html
+  /usr/share/doc/freetds-0.63/html/reference/a00399.html
+  /usr/share/doc/freetds-0.63/html/reference/annotated.html
+  /usr/share/doc/freetds-0.63/html/reference/bug.html
+  /usr/share/doc/freetds-0.63/html/reference/doxygen.css
+  /usr/share/doc/freetds-0.63/html/reference/doxygen.png
+  /usr/share/doc/freetds-0.63/html/reference/files.html
+  /usr/share/doc/freetds-0.63/html/reference/functions.html
+  /usr/share/doc/freetds-0.63/html/reference/globals.html
+  /usr/share/doc/freetds-0.63/html/reference/index.html
+  /usr/share/doc/freetds-0.63/html/reference/modules.html
+  /usr/share/doc/freetds-0.63/html/reference/pages.html
+  /usr/share/doc/freetds-0.63/html/reference/todo.html
+  /usr/share/doc/freetds-0.63/html/userguide/about.htm
+  /usr/share/doc/freetds-0.63/html/userguide/aboutunicode.htm
+  /usr/share/doc/freetds-0.63/html/userguide/acknowledgments.htm
+  /usr/share/doc/freetds-0.63/html/userguide/advocacy.htm
+  /usr/share/doc/freetds-0.63/html/userguide/apireference.htm
+  /usr/share/doc/freetds-0.63/html/userguide/appendmode.htm
+  /usr/share/doc/freetds-0.63/html/userguide/askingforhelp.htm
+  /usr/share/doc/freetds-0.63/html/userguide/build.htm
+  /usr/share/doc/freetds-0.63/html/userguide/choosingtdsprotocol.htm
+  /usr/share/doc/freetds-0.63/html/userguide/config.htm
+  /usr/share/doc/freetds-0.63/html/userguide/configs.htm
+  /usr/share/doc/freetds-0.63/html/userguide/configurations.htm
+  /usr/share/doc/freetds-0.63/html/userguide/confirminstall.htm
+  /usr/share/doc/freetds-0.63/html/userguide/contrib.htm
+  /usr/share/doc/freetds-0.63/html/userguide/contributors.htm
+  /usr/share/doc/freetds-0.63/html/userguide/ctlib.api.summary.htm
+  /usr/share/doc/freetds-0.63/html/userguide/dblib.api.summary.htm
+  /usr/share/doc/freetds-0.63/html/userguide/domains.htm
+  /usr/share/doc/freetds-0.63/html/userguide/dsnless.htm
+  /usr/share/doc/freetds-0.63/html/userguide/envvar.htm
+  /usr/share/doc/freetds-0.63/html/userguide/freetdsconf.htm
+  /usr/share/doc/freetds-0.63/html/userguide/freetdshistory.htm
+  /usr/share/doc/freetds-0.63/html/userguide/gfdl-1.htm
+  /usr/share/doc/freetds-0.63/html/userguide/gfdl-10.htm
+  /usr/share/doc/freetds-0.63/html/userguide/gfdl-11.htm
+  /usr/share/doc/freetds-0.63/html/userguide/gfdl-2.htm
+  /usr/share/doc/freetds-0.63/html/userguide/gfdl-3.htm
+  /usr/share/doc/freetds-0.63/html/userguide/gfdl-4.htm
+  /usr/share/doc/freetds-0.63/html/userguide/gfdl-5.htm
+  /usr/share/doc/freetds-0.63/html/userguide/gfdl-6.htm
+  /usr/share/doc/freetds-0.63/html/userguide/gfdl-7.htm
+  /usr/share/doc/freetds-0.63/html/userguide/gfdl-8.htm
+  /usr/share/doc/freetds-0.63/html/userguide/gfdl-9.htm
+  /usr/share/doc/freetds-0.63/html/userguide/gfdl.htm
+  /usr/share/doc/freetds-0.63/html/userguide/help.htm
+  /usr/share/doc/freetds-0.63/html/userguide/index.htm
+  /usr/share/doc/freetds-0.63/html/userguide/index.html
+  /usr/share/doc/freetds-0.63/html/userguide/install.htm
+  /usr/share/doc/freetds-0.63/html/userguide/interfacesfile.htm
+  /usr/share/doc/freetds-0.63/html/userguide/interfacesformat.htm
+  /usr/share/doc/freetds-0.63/html/userguide/interfaceslocation.htm
+  /usr/share/doc/freetds-0.63/html/userguide/interfacespurpose.htm
+  /usr/share/doc/freetds-0.63/html/userguide/iso8859.htm
+  /usr/share/doc/freetds-0.63/html/userguide/locales.htm
+  /usr/share/doc/freetds-0.63/html/userguide/logging.htm
+  /usr/share/doc/freetds-0.63/html/userguide/mailinglist.htm
+  /usr/share/doc/freetds-0.63/html/userguide/nonwestern.htm
+  /usr/share/doc/freetds-0.63/html/userguide/odbc.api.summary.htm
+  /usr/share/doc/freetds-0.63/html/userguide/odbcconnattr.htm
+  /usr/share/doc/freetds-0.63/html/userguide/odbcdiagnose.htm
+  /usr/share/doc/freetds-0.63/html/userguide/odbcinionly.htm
+  /usr/share/doc/freetds-0.63/html/userguide/odbcombo.htm
+  /usr/share/doc/freetds-0.63/html/userguide/osissues.htm
+  /usr/share/doc/freetds-0.63/html/userguide/packages.htm
+  /usr/share/doc/freetds-0.63/html/userguide/pagenodata.htm
+  /usr/share/doc/freetds-0.63/html/userguide/perl.htm
+  /usr/share/doc/freetds-0.63/html/userguide/php.htm
+  /usr/share/doc/freetds-0.63/html/userguide/prepodbc.htm
+  /usr/share/doc/freetds-0.63/html/userguide/programming.htm
+  /usr/share/doc/freetds-0.63/html/userguide/projects.htm
+  /usr/share/doc/freetds-0.63/html/userguide/python.htm
+  /usr/share/doc/freetds-0.63/html/userguide/samplecode.htm
+  /usr/share/doc/freetds-0.63/html/userguide/seemtooslow.htm
+  /usr/share/doc/freetds-0.63/html/userguide/serverthere.htm
+  /usr/share/doc/freetds-0.63/html/userguide/software.htm
+  /usr/share/doc/freetds-0.63/html/userguide/stunnel.htm
+  /usr/share/doc/freetds-0.63/html/userguide/sybsql.htm
+  /usr/share/doc/freetds-0.63/html/userguide/tdshistory.htm
+  /usr/share/doc/freetds-0.63/html/userguide/tdspool.htm
+  /usr/share/doc/freetds-0.63/html/userguide/troubleshooting.htm
+  /usr/share/doc/freetds-0.63/html/userguide/unicode.htm
+  /usr/share/doc/freetds-0.63/html/userguide/unicodefreetds.htm
+  /usr/share/doc/freetds-0.63/html/userguide/unicodegoodbad.htm
+  /usr/share/doc/freetds-0.63/html/userguide/unicodeutf.htm
+  /usr/share/doc/freetds-0.63/html/userguide/what.htm
+
+------------------
+
+Port Notes:
+
+----- version 0.63-1 -----
+Initial release for Cygwin.
+
+
+Cygwin port maintained by:
+http://cygwinports.dotsrc.org/
diff -urN -x .build -x .inst -x .sinst -x configure -x 'Makefile.in*' -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x 'mkinstalldirs*' -x autom4te.cache -x '*compile' freetds-0.63-orig/CYGWIN-PATCHES/devel.hint freetds-0.63/CYGWIN-PATCHES/devel.hint
--- freetds-0.63-orig/CYGWIN-PATCHES/devel.hint	1969-12-31 18:00:00.000000000 -0600
+++ freetds-0.63/CYGWIN-PATCHES/devel.hint	2005-11-14 17:51:26.812500000 -0600
@@ -0,0 +1,12 @@
+category: Database Devel
+requires: cygwin libiconv
+external-source: freetds
+sdesc: "Library for talking to MS SQL and Sybase databases (development)"
+ldesc: "FreeTDS is a set of libraries for Unix and Linux that allows your
+programs to natively talk to Microsoft SQL Server and Sybase databases.
+
+Technically speaking, FreeTDS is an open source implementation of the
+TDS (Tabular Data Stream) protocol used by these databases for their own
+clients. It supports many different flavors of the protocol and three
+APIs to access it. Additionally FreeTDS works with other software such
+as Perl and PHP, providing access from those languages as well."
diff -urN -x .build -x .inst -x .sinst -x configure -x 'Makefile.in*' -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x 'mkinstalldirs*' -x autom4te.cache -x '*compile' freetds-0.63-orig/CYGWIN-PATCHES/doc.hint freetds-0.63/CYGWIN-PATCHES/doc.hint
--- freetds-0.63-orig/CYGWIN-PATCHES/doc.hint	1969-12-31 18:00:00.000000000 -0600
+++ freetds-0.63/CYGWIN-PATCHES/doc.hint	2005-11-14 17:51:26.859375000 -0600
@@ -0,0 +1,11 @@
+category: Database Doc
+external-source: freetds
+sdesc: "Library for talking to MS SQL and Sybase databases (documentation)"
+ldesc: "FreeTDS is a set of libraries for Unix and Linux that allows your
+programs to natively talk to Microsoft SQL Server and Sybase databases.
+
+Technically speaking, FreeTDS is an open source implementation of the
+TDS (Tabular Data Stream) protocol used by these databases for their own
+clients. It supports many different flavors of the protocol and three
+APIs to access it. Additionally FreeTDS works with other software such
+as Perl and PHP, providing access from those languages as well."
diff -urN -x .build -x .inst -x .sinst -x configure -x 'Makefile.in*' -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x 'mkinstalldirs*' -x autom4te.cache -x '*compile' freetds-0.63-orig/CYGWIN-PATCHES/libct.hint freetds-0.63/CYGWIN-PATCHES/libct.hint
--- freetds-0.63-orig/CYGWIN-PATCHES/libct.hint	1969-12-31 18:00:00.000000000 -0600
+++ freetds-0.63/CYGWIN-PATCHES/libct.hint	2005-11-14 17:51:26.906250000 -0600
@@ -0,0 +1,12 @@
+category: Database Libs
+requires: cygwin libiconv2
+external-source: freetds
+sdesc: "Library for talking to MS SQL and Sybase databases (CT-lib)"
+ldesc: "FreeTDS is a set of libraries for Unix and Linux that allows your
+programs to natively talk to Microsoft SQL Server and Sybase databases.
+
+Technically speaking, FreeTDS is an open source implementation of the
+TDS (Tabular Data Stream) protocol used by these databases for their own
+clients. It supports many different flavors of the protocol and three
+APIs to access it. Additionally FreeTDS works with other software such
+as Perl and PHP, providing access from those languages as well."
diff -urN -x .build -x .inst -x .sinst -x configure -x 'Makefile.in*' -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x 'mkinstalldirs*' -x autom4te.cache -x '*compile' freetds-0.63-orig/CYGWIN-PATCHES/libsybdb.hint freetds-0.63/CYGWIN-PATCHES/libsybdb.hint
--- freetds-0.63-orig/CYGWIN-PATCHES/libsybdb.hint	1969-12-31 18:00:00.000000000 -0600
+++ freetds-0.63/CYGWIN-PATCHES/libsybdb.hint	2005-11-14 17:51:26.921875000 -0600
@@ -0,0 +1,12 @@
+category: Database Libs
+requires: cygwin libiconv2
+external-source: freetds
+sdesc: "Library for talking to MS SQL and Sybase databases (Sybase lib)"
+ldesc: "FreeTDS is a set of libraries for Unix and Linux that allows your
+programs to natively talk to Microsoft SQL Server and Sybase databases.
+
+Technically speaking, FreeTDS is an open source implementation of the
+TDS (Tabular Data Stream) protocol used by these databases for their own
+clients. It supports many different flavors of the protocol and three
+APIs to access it. Additionally FreeTDS works with other software such
+as Perl and PHP, providing access from those languages as well."
diff -urN -x .build -x .inst -x .sinst -x configure -x 'Makefile.in*' -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x 'mkinstalldirs*' -x autom4te.cache -x '*compile' freetds-0.63-orig/CYGWIN-PATCHES/libtds.hint freetds-0.63/CYGWIN-PATCHES/libtds.hint
--- freetds-0.63-orig/CYGWIN-PATCHES/libtds.hint	1969-12-31 18:00:00.000000000 -0600
+++ freetds-0.63/CYGWIN-PATCHES/libtds.hint	2005-11-14 17:51:26.937500000 -0600
@@ -0,0 +1,12 @@
+category: Database Libs
+requires: cygwin libiconv2
+external-source: freetds
+sdesc: "Library for talking to MS SQL and Sybase databases (core lib)"
+ldesc: "FreeTDS is a set of libraries for Unix and Linux that allows your
+programs to natively talk to Microsoft SQL Server and Sybase databases.
+
+Technically speaking, FreeTDS is an open source implementation of the
+TDS (Tabular Data Stream) protocol used by these databases for their own
+clients. It supports many different flavors of the protocol and three
+APIs to access it. Additionally FreeTDS works with other software such
+as Perl and PHP, providing access from those languages as well."
diff -urN -x .build -x .inst -x .sinst -x configure -x 'Makefile.in*' -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x 'mkinstalldirs*' -x autom4te.cache -x '*compile' freetds-0.63-orig/CYGWIN-PATCHES/libtdssrv.hint freetds-0.63/CYGWIN-PATCHES/libtdssrv.hint
--- freetds-0.63-orig/CYGWIN-PATCHES/libtdssrv.hint	1969-12-31 18:00:00.000000000 -0600
+++ freetds-0.63/CYGWIN-PATCHES/libtdssrv.hint	2005-11-14 17:51:26.953125000 -0600
@@ -0,0 +1,12 @@
+category: Database Libs
+requires: cygwin libiconv2
+external-source: freetds
+sdesc: "Library for talking to MS SQL and Sybase databases (server lib)"
+ldesc: "FreeTDS is a set of libraries for Unix and Linux that allows your
+programs to natively talk to Microsoft SQL Server and Sybase databases.
+
+Technically speaking, FreeTDS is an open source implementation of the
+TDS (Tabular Data Stream) protocol used by these databases for their own
+clients. It supports many different flavors of the protocol and three
+APIs to access it. Additionally FreeTDS works with other software such
+as Perl and PHP, providing access from those languages as well."
diff -urN -x .build -x .inst -x .sinst -x configure -x 'Makefile.in*' -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x 'mkinstalldirs*' -x autom4te.cache -x '*compile' freetds-0.63-orig/CYGWIN-PATCHES/setup.hint freetds-0.63/CYGWIN-PATCHES/setup.hint
--- freetds-0.63-orig/CYGWIN-PATCHES/setup.hint	1969-12-31 18:00:00.000000000 -0600
+++ freetds-0.63/CYGWIN-PATCHES/setup.hint	2005-11-14 17:51:26.984375000 -0600
@@ -0,0 +1,12 @@
+category: Database
+requires: cygwin libiconv2 libncurses8 libreadline6 libsybdb5 libtbssrv2
+external-source: freetds
+sdesc: "Library for talking to MS SQL and Sybase databases (utilities)"
+ldesc: "FreeTDS is a set of libraries for Unix and Linux that allows your
+programs to natively talk to Microsoft SQL Server and Sybase databases.
+
+Technically speaking, FreeTDS is an open source implementation of the
+TDS (Tabular Data Stream) protocol used by these databases for their own
+clients. It supports many different flavors of the protocol and three
+APIs to access it. Additionally FreeTDS works with other software such
+as Perl and PHP, providing access from those languages as well."
diff -urN -x .build -x .inst -x .sinst -x configure -x 'Makefile.in*' -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x 'mkinstalldirs*' -x autom4te.cache -x '*compile' freetds-0.63-orig/configure.in freetds-0.63/configure.in
--- freetds-0.63-orig/configure.in	2005-03-29 18:42:21.000000000 -0600
+++ freetds-0.63/configure.in	2005-11-14 12:24:12.281250000 -0600
@@ -350,6 +350,21 @@
 	odbc=true
 fi
 
+AC_ARG_WITH(odbc32,
+AC_HELP_STRING([--with-odbc32=DIR], [build odbc driver against ODBC32 in DIR]))
+if test "$with_odbc32"; then
+	if echo "$with_odbc32" | grep -v '^/'; then
+		with_unixodbc="$PWD/$with_unixodbc"
+	fi
+	CPPFLAGS="$CPPFLAGS -DUNIXODBC"
+	ODBC_INC="-I$with_unixodbc/include"
+	LDFLAGS="$LDFLAGS -L$with_unixodbc/lib"
+	ODBCLIB="-lodbc32"
+	odbc=true
+	unixodbc=true
+	odbc32=true
+fi
+
 build_odbc=yes
 if test "$odbc" != "true"; then
 AC_ARG_ENABLE(odbc,
@@ -370,12 +385,22 @@
 		[AC_CHECK_LIB(odbc, SQLConnect, [AC_CHECK_HEADER([sql.h], [odbc=true
 	unixodbc=true
 	CPPFLAGS="$CPPFLAGS -DUNIXODBC"
-	ODBCLIB="-lodbc"])] )] )
+	ODBCLIB="-lodbc"])],
+		[AC_CHECK_LIB(odbc32, SQLConnect, [AC_CHECK_HEADER([sql.h], [odbc=true
+	unixodbc=true
+	odbc32=true
+	CPPFLAGS="$CPPFLAGS -DUNIXODBC"
+	ODBCLIB="-lodbc32"], ,[#include <windows.h>])], ,[-Wl,--enable-runtime-pseudo-reloc])] )] )
 	fi
 	if test "$odbc" = "true"; then
 		if test "$unixodbc" = "true"; then
-			AC_CHECK_LIB(odbcinst, SQLGetPrivateProfileString, [ODBCINSTLIB="-lodbcinst"
+			if test "$odbc32" = "true"; then
+				ODBCINSTLIB="-lodbccp32"
+				AC_DEFINE_UNQUOTED(HAVE_SQLGETPRIVATEPROFILESTRING, 1, [Define to 1 if you have the SQLGetPrivateProfileString function.])
+			else
+				AC_CHECK_LIB(odbcinst, SQLGetPrivateProfileString, [ODBCINSTLIB="-lodbcinst"
 	AC_DEFINE_UNQUOTED(HAVE_SQLGETPRIVATEPROFILESTRING, 1, [Define to 1 if you have the SQLGetPrivateProfileString function.])])
+			fi
 		elif test "$iodbc" = "true"; then
 			AC_CHECK_LIB(iodbcinst, SQLGetPrivateProfileString, [ODBCINSTLIB="-liodbcinst"
 	AC_DEFINE_UNQUOTED(HAVE_SQLGETPRIVATEPROFILESTRING, 1, [Define to 1 if you have the SQLGetPrivateProfileString function.])])
@@ -393,9 +418,14 @@
 	AC_TRY_COMPILE([#include <sqltypes.h>], [SQLLEN len = 123;],
 		AC_DEFINE(HAVE_SQLLEN, 1, [Define if sqltypes.h define SQLLEN])
 		# check SQLColAttribute definition
-		AC_TRY_COMPILE([#include <sqltypes.h>
+		AC_TRY_COMPILE([
+#ifdef __CYGWIN__
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
+#endif
 #include <sql.h>
 #include <sqlext.h>
+#include <sqltypes.h>
 
 SQLRETURN  SQL_API SQLColAttribute (SQLHSTMT StatementHandle,
 	SQLUSMALLINT ColumnNumber, SQLUSMALLINT FieldIdentifier,
@@ -411,6 +441,7 @@
 
 AM_CONDITIONAL(ODBC, test "$odbc" = "true")
 AM_CONDITIONAL(UNIXODBC, test "$unixodbc" = "true")
+AM_CONDITIONAL(ODBC32, test "$odbc32" = "true")
 AC_SUBST(ODBC)
 AC_SUBST(ODBC_INC)
 AC_SUBST(ODBCLIB)
diff -urN -x .build -x .inst -x .sinst -x configure -x 'Makefile.in*' -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x 'mkinstalldirs*' -x autom4te.cache -x '*compile' freetds-0.63-orig/include/tdsodbc.h freetds-0.63/include/tdsodbc.h
--- freetds-0.63-orig/include/tdsodbc.h	2004-10-28 07:42:12.000000000 -0500
+++ freetds-0.63/include/tdsodbc.h	2005-11-13 19:45:15.359375000 -0600
@@ -23,6 +23,10 @@
 #include <tds.h>
 
 #ifdef UNIXODBC
+#ifdef __CYGWIN__
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
+#endif
 #include <sql.h>
 #include <sqlext.h>
 #include <odbcinst.h>
diff -urN -x .build -x .inst -x .sinst -x configure -x 'Makefile.in*' -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x 'mkinstalldirs*' -x autom4te.cache -x '*compile' freetds-0.63-orig/src/ctlib/Makefile.am freetds-0.63/src/ctlib/Makefile.am
--- freetds-0.63-orig/src/ctlib/Makefile.am	2004-12-09 06:54:43.000000000 -0600
+++ freetds-0.63/src/ctlib/Makefile.am	2005-11-13 18:34:39.765625000 -0600
@@ -6,7 +6,7 @@
 
 # version bumped for 0.63, I hope we'll bump it less in the future -- freddy77
 ##libct_la_LDFLAGS=	-version-info 3:0:0 -export-symbols-regex '^(cs_|ct_|blk_).*' -Wl,-Bsymbolic
-libct_la_LDFLAGS=	-version-info 3:0:0 -export-symbols-regex '^(cs_|ct_|blk_).*' $(FREETDS_SYMBOLIC)
+libct_la_LDFLAGS=	-version-info 3:0:0 -export-symbols-regex '^(cs_|ct_|blk_).*' $(FREETDS_SYMBOLIC) -no-undefined
 
 libct_la_LIBADD	=	../tds/libtds_objects.la ../replacements/libreplacements.la $(NETWORK_LIBS) $(LIBICONV) $(FREETDS_LIBGCC)
 
diff -urN -x .build -x .inst -x .sinst -x configure -x 'Makefile.in*' -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x 'mkinstalldirs*' -x autom4te.cache -x '*compile' freetds-0.63-orig/src/dblib/Makefile.am freetds-0.63/src/dblib/Makefile.am
--- freetds-0.63-orig/src/dblib/Makefile.am	2004-12-09 06:54:45.000000000 -0600
+++ freetds-0.63/src/dblib/Makefile.am	2005-11-13 18:36:21.015625000 -0600
@@ -8,7 +8,7 @@
 EXTRA_DIST	=	dbopen.c
 libsybdb_la_SOURCES=	dblib.c dbutil.c rpc.c bcp.c xact.c $(DBOPEN)
 # version bumped for 0.63, I hope we'll bump it less in the future -- freddy77
-libsybdb_la_LDFLAGS=	-version-info 5:0:0 -export-symbols-regex '^(db|bcp_|tdsdump_open|tdsdbopen|.*_xact|close_commit|open_commit|.?asprintf).*' $(FREETDS_SYMBOLIC)
+libsybdb_la_LDFLAGS=	-version-info 5:0:0 -export-symbols-regex '^(db|bcp_|tdsdump_open|tdsdbopen|.*_xact|close_commit|open_commit|.?asprintf).*' $(FREETDS_SYMBOLIC) -no-undefined
 ##libsybdb_la_LDFLAGS=	-version-info 5:0:0 -export-symbols-regex '^(db|bcp_|tdsdump_open|tdsdbopen|.*_xact|close_commit|open_commit).*' -Wl,-Bsymbolic
 libsybdb_la_LIBADD=	../tds/libtds_objects.la ../replacements/libreplacements.la $(NETWORK_LIBS) $(LIBICONV) $(FREETDS_LIBGCC)
 
diff -urN -x .build -x .inst -x .sinst -x configure -x 'Makefile.in*' -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x 'mkinstalldirs*' -x autom4te.cache -x '*compile' freetds-0.63-orig/src/odbc/Makefile.am freetds-0.63/src/odbc/Makefile.am
--- freetds-0.63-orig/src/odbc/Makefile.am	2004-12-09 06:54:46.000000000 -0600
+++ freetds-0.63/src/odbc/Makefile.am	2005-11-14 13:20:54.359375000 -0600
@@ -1,7 +1,8 @@
 # $Id: freetds-0.63-1.patch,v 1.1.1.1 2006-06-16 19:01:04 yselkowitz Exp $
 SUBDIRS		= unittests
 
-lib_LTLIBRARIES	=	libtdsodbc.la
+odbcdir = $(libdir)/odbc
+odbc_LTLIBRARIES	=	libtdsodbc.la
 ##EXTRA_LTLIBRARIES	=	libtdsodbc.la
 libtdsodbc_la_SOURCES=	odbc.c connectparams.c convert_tds2sql.c \
 	descriptor.c prepare_query.c odbc_util.c \
@@ -12,7 +13,7 @@
 libtdsodbc_la_LIBADD=	../tds/libtds_objects.la ../replacements/libreplacements.la $(ODBCINSTLIB) $(NETWORK_LIBS) $(LIBICONV) $(FREETDS_LIBGCC)
 INCLUDES	=	-I$(top_srcdir)/include $(ODBC_INC)
 # -module is needed by Darwin (Mac OS X)
-libtdsodbc_la_LDFLAGS = -export-symbols-regex '^(SQL|ODBCINST).*' $(FREETDS_SYMBOLIC) -module
+libtdsodbc_la_LDFLAGS = -export-symbols-regex '^(SQL|ODBCINST).*' $(FREETDS_SYMBOLIC) -module -avoid-version -no-undefined
 
 ## Need blank statement to avoid compiling odbc.c
 odbc: $(EXTRA_LTLIBRARIES)
diff -urN -x .build -x .inst -x .sinst -x configure -x 'Makefile.in*' -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x 'mkinstalldirs*' -x autom4te.cache -x '*compile' freetds-0.63-orig/src/odbc/odbc.c freetds-0.63/src/odbc/odbc.c
--- freetds-0.63-orig/src/odbc/odbc.c	2004-10-28 07:42:12.000000000 -0500
+++ freetds-0.63/src/odbc/odbc.c	2005-11-14 12:51:27.593750000 -0600
@@ -358,7 +358,7 @@
 
 	/* add login info */
 	if (hwnd) {
-#ifdef WIN32
+#if defined(WIN32) && !defined(__CYGWIN__)
 		/* prompt for login information */
 		if (!get_login_info(hwnd, connection)) {
 			tds_free_connection(connection);
diff -urN -x .build -x .inst -x .sinst -x configure -x 'Makefile.in*' -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x 'mkinstalldirs*' -x autom4te.cache -x '*compile' freetds-0.63-orig/src/replacements/vasprintf.c freetds-0.63/src/replacements/vasprintf.c
--- freetds-0.63-orig/src/replacements/vasprintf.c	2004-03-15 03:15:44.000000000 -0600
+++ freetds-0.63/src/replacements/vasprintf.c	2005-11-14 12:53:04.953125000 -0600
@@ -84,9 +84,9 @@
 
 #ifdef _REENTRANT
 
-# ifdef WIN32
+#if defined(WIN32) && !defined(__CYGWIN__)
 #  error Win32 do not have /dev/null, should use vsnprintf version
-# endif
+#endif
 
 	if ((fp = fopen(_PATH_DEVNULL, "w")) == NULL)
 		return -1;
diff -urN -x .build -x .inst -x .sinst -x configure -x 'Makefile.in*' -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x 'mkinstalldirs*' -x autom4te.cache -x '*compile' freetds-0.63-orig/src/server/Makefile.am freetds-0.63/src/server/Makefile.am
--- freetds-0.63-orig/src/server/Makefile.am	2004-12-12 18:15:08.000000000 -0600
+++ freetds-0.63/src/server/Makefile.am	2005-11-13 18:37:43.203125000 -0600
@@ -2,7 +2,7 @@
 lib_LTLIBRARIES	=	libtdssrv.la
 libtdssrv_la_SOURCES=	query.c server.c login.c
 # bump server version whenever bumping libtds version
-libtdssrv_la_LDFLAGS=	-version-info 2:0:0
+libtdssrv_la_LDFLAGS=	-version-info 2:0:0 -no-undefined
 libtdssrv_la_LIBADD =	../tds/libtds_objects.la ../replacements/libreplacements.la $(NETWORK_LIBS) $(LIBICONV) $(FREETDS_LIBGCC)
 INCLUDES	=	-I$(top_srcdir)/include
 noinst_PROGRAMS	= tdssrv
diff -urN -x .build -x .inst -x .sinst -x configure -x 'Makefile.in*' -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x 'mkinstalldirs*' -x autom4te.cache -x '*compile' freetds-0.63-orig/src/tds/Makefile.am freetds-0.63/src/tds/Makefile.am
--- freetds-0.63-orig/src/tds/Makefile.am	2004-12-12 18:19:05.000000000 -0600
+++ freetds-0.63/src/tds/Makefile.am	2005-11-13 18:38:00.921875000 -0600
@@ -14,7 +14,7 @@
 lib_LTLIBRARIES		=	libtds.la
 libtds_la_SOURCES	=
 # Release 0.63 is version 4, cf. ML 30 Nov 2004 11:08:28 -0600
-libtds_la_LDFLAGS	= -version-info 4:0:0 -export-symbols-regex '^(tds_|tdsdump_|tds5_|tds7_).*'
+libtds_la_LDFLAGS	= -version-info 4:0:0 -export-symbols-regex '^(tds_|tdsdump_|tds5_|tds7_).*' -no-undefined
 # 			bump server version whenever bumping libtds version
 libtds_la_LIBADD 	= libtds_objects.la ../replacements/libreplacements.la $(NETWORK_LIBS) $(LIBICONV) $(FREETDS_LIBGCC)
 noinst_HEADERS		= tds_willconvert.h character_sets.h \
diff -urN -x .build -x .inst -x .sinst -x configure -x 'Makefile.in*' -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x 'mkinstalldirs*' -x autom4te.cache -x '*compile' freetds-0.63-orig/src/tds/threadsafe.c freetds-0.63/src/tds/threadsafe.c
--- freetds-0.63-orig/src/tds/threadsafe.c	2004-05-03 15:02:47.000000000 -0500
+++ freetds-0.63/src/tds/threadsafe.c	2005-11-14 12:53:32.890625000 -0600
@@ -68,7 +68,7 @@
 #include <netinet/in.h>
 #endif /* HAVE_NETINET_IN_H */
 
-#ifdef WIN32
+#if defined(WIN32) && !defined(__CYGWIN__)
 #include <shlobj.h>
 #endif
 
